### PR TITLE
feat: add expression helpers for parsing LoadSubsetOptions in queryFn

### PR DESCRIPTION
**Pull request by KyleAMathews**
## Summary

Adds reusable expression parsing utilities to help developers translate TanStack DB predicates (where, orderBy, limit) into their API's format when implementing query collections with predicate push-down.

### New Expression Helpers (`packages/query-db-collection/src/expression-helpers.ts`)

- **`parseWhereExpression`**: Parse where clauses with custom handlers for each operator
- **`parseOrderByExpression`**: Parse order by into simple array format
- **`extractSimpleComparisons`**: Extract simple AND-ed filters for basic use cases
- **`parseLoadSubsetOptions`**: Convenience function to parse all options at once
- **`walkExpression`**, **`extractFieldPath`**, **`extractValue`**: Lower-level helpers for custom parsing

### Documentation

Added comprehensive section to `docs/collections/query-collection.md`:
- How LoadSubsetOptions are passed via `ctx.meta.loadSubsetOptions`
- Expression helper usage with REST API examples
- GraphQL example
- API reference for all helper functions
- Query key builder pattern
- Tips and best practices

### Tests

35 test cases with 93.65% coverage:
- All helper functions tested
- Various operator combinations (eq, lt, gt, and, or, etc.)
- Edge cases (null/undefined, nested expressions)
- Integration test with complex real-world query

## Why This Matters

Without these helpers, developers had to manually parse expression AST trees:

```typescript
// Before - complex and error-prone
if (where && where.type === 'func' && where.name === 'eq') {
  const field = where.args[0].path[0]
  const value = where.args[1].value
  // ... more traversal logic
}
```

With helpers, it's straightforward:

```typescript
// After - simple and maintainable
const filters = parseWhereExpression(where, {
  handlers: {
    eq: (field, value) => ({ [field.join('.')]: value }),
    lt: (field, value) => ({ [`${field.join('.')}_lt`]: value }),
    and: (...conditions) => Object.assign({}, ...conditions)
  }
})
```

## Example Usage

```typescript
queryFn: async (ctx) => {
  const { where, orderBy, limit } = ctx.meta.loadSubsetOptions

  // Parse into your API format
  const parsed = parseLoadSubsetOptions({ where, orderBy, limit })

  // Build API request
  const params = new URLSearchParams()
  parsed.filters.forEach(({ field, operator, value }) => {
    if (operator === 'eq') {
      params.set(field.join('.'), String(value))
    }
  })

  return fetch(`/api/products?${params}`).then(r => r.json())
}
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/db/src/query/expression-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+   * - gte: greater than or equal (>=)
+   * - lt: less than (<)
+   * - lte: less than or equal (<=)
+   * - and: logical AND
+   * - or: logical OR
+   * - in: IN clause
+   */
+  handlers: {
+    [operator: string]: (...args: Array<any>) => T
+  }
```
Comment (samwillis) [inline packages/db/src/query/expression-helpers.ts:71]: can we improve the type here such that it knows the operators that the IR supports?

File: packages/query-db-collection/src/expression-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+import type { IR } from "@tanstack/db"
+
+type BasicExpression<T = any> = IR.BasicExpression<T>
+type OrderBy = IR.OrderBy
+
+/**
+ * Represents a simple field path extracted from an expression
+ */
+export type FieldPath = Array<string>
```
Comment (samwillis) [inline packages/query-db-collection/src/expression-helpers.ts]: We may want to support accessing elements in a array:

```suggestion
export type FieldPath = Array<string | number>
```

File: packages/db/src/query/expression-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+ * @example
+ * ```typescript
+ * const comparisons = extractSimpleComparisons(where)
+ * // Returns: [
+ * //   { field: ['category'], operator: 'eq', value: 'electronics' },
+ * //   { field: ['price'], operator: 'lt', value: 100 }
+ * // ]
+ * ```
+ */
+export function extractSimpleComparisons(
```
Comment (samwillis) [inline packages/db/src/query/expression-helpers.ts:323]: We may want this to throw if if finds something it can't extract such as an `or`

File: packages/query-db-collection/src/expression-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+      throw new Error(
+        `ORDER BY expression must be a field reference, got: ${clause.expression.type}`
+      )
+    }
+
+    return {
+      field,
+      direction: clause.compareOptions.direction,
+      nulls: clause.compareOptions.nulls,
+    }
```
Comment (samwillis) [inline packages/query-db-collection/src/expression-helpers.ts]: we need to return the local / string comparison options too

**Comment by samwillis** [comment]
Mostly there except for the above comments. I will rebase #773 on this and use it.

**Comment by samwillis** [comment]
@KyleAMathews there are a few tests failing too, looks like just types

**Comment by KyleAMathews** [comment]
Cool! BTW, I've been thinking we'll probably want to move something like these utilities into a shared package as other collections will surely want to do something similar. The babel of the IR haha.
